### ISSUE TITLE
Using YOLOv8 on large remote sensing satellite images

### ISSUE THREAD

**Issue by SyedAhmed68** [ultralytics/ultralytics#17193]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I am using YOLOv8 for object detection on a high-resolution satellite imagery. The original image is quite large (approx. 6500px*5300px). 
my first question is, **is there any way to run a trained YOLOv8 model and get results without having to resize/clip the large image?**

For my training part, I have used a smaller subset of the large image. The size is 2056*1903px, further I have made smaller 128*128 px image chips out of the subset then annotated them and then trained the model.
When I run the trained model on 128*128 size image chips, the model performs wonderfully. But when I use a larger image i.e. 2056*1903px, then the model can not detect anything accurately. 
**Do you have any suggestions what should be the best approach here**?

Finally, after running the model, I get results in two formats 1) jpeg images with bounding box and 2) a txt file that contains the bounding box information. But for the nature of my work, I need a a shape file layer that will contain all the bounding box so that I can create a full map. I tried to using merge all the txt files information and produce a shape file but nothing rally works in this case. So **can you please suggest me any approach by which I can get the bounding boxes as a shape file layer**?

I will be waiting for your feedback. Thank you very much for your time, consideration and suggestions.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @SyedAhmed68, thank you for reaching out to the Ultralytics community with your questions on YOLOv8 and high-resolution satellite imagery 🌍🚀! This is an automated response to help guide you through some resources while an Ultralytics engineer will also assist soon.

For users working with high-resolution images, creating smaller image chips for model training as you've done is a common practice. If you are curious about running YOLOv8 without resizing, we'd encourage sharing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to better understand and debug any issues.

To enhance model performance on larger images, consider the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) which can greatly impact detection accuracy.

As for exporting your results to a shapefile layer, while our package natively provides JPEG and TXT outputs, transforming these into GIS formats like shapefiles often involves additional processing. You might find community channels helpful for sharing such methods.

Explore the Ultralytics community to gain insights and get support:
- Join our [Discord](https://ultralytics.com/discord) 🎧 for real-time discussions.
- Dive into thoughtful exchanges on our [Discourse](https://community.ultralytics.com).
- Engage with other users on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you're using the latest `ultralytics` package version to avoid potential bugs that might have been resolved. Follow these upgrade instructions:

```bash
pip install -U ultralytics
```

## Environments

Consider verified environments to efficiently run YOLO with all necessary dependencies in place:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Refer to our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Check out our [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Review our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

Follow the latest [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests to verify functionality across various operating systems.

Feel free to provide further details or examples in your issue to help us resolve it efficiently. Thank you for your patience and enthusiasm in working with Ultralytics! 🎉

**Comment by Y-T-G** [comment]
You can try SAHI

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/6929

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in the linked PR #20223, which introduces comprehensive multispectral image support throughout the Ultralytics ecosystem.

Key changes include:
- Full support for multi-channel datasets. You can now specify the number of channels in your dataset config, enabling seamless training, validation, prediction, and export with images that have more than 3 channels (such as 10-channel multispectral data).
- All YOLO model trainers, augmentations, and preprocessing steps now dynamically adjust to the dataset’s channel count.
- Model export and inference pipelines have been updated to handle and record the correct number of input channels, with metadata included for exported models.
- Visualization functions now gracefully display multispectral images by using only the first three channels.
- Additional tests ensure reliable multi-channel handling across different workflows.

Special thanks to @Laughing-q for the main implementation and to the contributors @glenn-jocher and @Y-T-G for their support.

To test if this resolves your issue, you can install the latest update:
```
pip install -U ultralytics>=8.3.12
```
Alternatively, you can install directly from the main branch for the most recent changes:
```
pip install git+https://github.com/ultralytics/ultralytics.git@main
```
Or, if you prefer, you can wait for the next official PyPI release.

Please let us know if the changes in this update resolve your issue. Thank you for reporting this, and feel free to share any further feedback if you encounter any other problems! 🙏

**Comment by glenn-jocher** [comment]
Hi @SyedAhmed68, 

Working with large satellite imagery is definitely a specialized use case. Here's how to address your questions:

1. For large images, the most effective approach is to use a sliding window technique like SAHI (Slicing Aided Hyper Inference) as suggested. SAHI automatically handles slicing your large image, running inference on each slice, and then merging the results with proper coordinate transformation.

2. Your model performs well on 128×128 chips but fails on larger images likely because of scale discrepancy. This is common in satellite imagery. I recommend:
   - Training with multi-scale augmentation to help the model generalize
   - Consider using Oriented Bounding Box (OBB) detection which is specifically designed for satellite imagery: https://docs.ultralytics.com/tasks/obb/

3. For converting results to shapefiles, you can use libraries like GeoPandas to convert your detection results:
   ```python
   import geopandas as gpd
   from shapely.geometry import box
   
   # Create geometries from your bounding boxes
   geometries = [box(x1, y1, x2, y2) for x1, y1, x2, y2 in your_bounding_boxes]
   # Create GeoDataFrame
   gdf = gpd.GeoDataFrame(geometry=geometries)
   # Save as shapefile
   gdf.to_file("detections.shp")
   ```

You might also find our satellite image analysis documentation useful: https://www.ultralytics.com/glossary/satellite-image-analysis
